File: internal/game/scoreboard.go
package game

type ScoreBoard struct {
	Scores []Player `json:"scores"`
}

// DummyScoreBoard is a dummy scoreboard
var DummyScoreBoard = &ScoreBoard{
	Scores: []Player{
		{
			Name:  "Player 1",
			Score: 1,
		},
		{
			Name:  "Player 2",
			Score: 2,
		},
	},
}

// NewScoreBoard creates a new scoreboard. For now, it is a dummy with two players
func NewScoreBoard(_ string) *ScoreBoard {
	return DummyScoreBoard
}

File: frontend/src/components/scoreboard.js
import React, {useState, useEffect} from "react";
import axios from "axios";
import {Container, Table} from "react-bootstrap";

const PlayerRankList = () => {
    const [players, setPlayers] = useState({
        scores: [
            {
                id: 1,
                name: "John Doe",
                score: 100
            }
        ]
    });

    useEffect(() => {
        axios.get("/api/scoreboard")
            .then(res => {
                console.log(res.data);
                setPlayers(res.data);
            })
            .catch(
                err => console.log(err)
            )
    }, []);

    return (
        <Container>
            <Table striped bordered hover size="sm" variant="light" responsive className="opacity-75 text-opacity-100">
                <thead className="bg-primary">
                <tr>
                    <th>Rank</th>
                    <th>Name</th>
                    <th>Score</th>
                </tr>
                </thead>
                <tbody>
                {players.scores.map((player, index) => (
                    <tr key={player.id}>
                        <td>{index + 1}</td>
                        <td>{player.name}</td>
                        <td>{player.score}</td>
                    </tr>
                ))}
                </tbody>
            </Table>
        </Container>
    );
};

export default PlayerRankList;
